src/components/binding/binding-panel.js

```diff
@@ -213,13 +187,44 @@ genotet.BindingPanel.prototype.updateTracks = function() {
       this.container.find('#genes #track-' + index).remove();
     }
   }
+
+  this.signal('loadBindingList');
+
+  this.container.find('#genes .glyphicon-remove')
+    .css('display', this.data.tracks.length == 1 ? 'none' : '');
+};
+
+/**
+ * Updates the track list for panel.
+ */
+genotet.BindingPanel.prototype.updateTrackList = function() {
```

Comment:
I don't think I'm going in loops anyway. You have misuderstood the keypoint of this entire change. The change is to make your panel update logic clean and neat. However rightnow it's still a bit chaos.

**updateTracks: Updates the gene selected for each track**
But in fact it is also removing track DOMs, as well as loading binding list.

Once again, what I suggest is:
Create a uniform entry for panel tracks update. Let's call it update(). Whenever you want to redraw the panel. You called panel.update(). Within the update(), you may want to retrieve the new binding list. So you can first retrieve the list and then update the DOMs. The steps of update() are thus:
- Retrieve the binding list
- Do proper track addition/removal when the number of tracks change
- Update the selected item of each track